### PR TITLE
handbook: add Go test naming convention

### PR DESCRIPTION
**Pull request by rvantonder**
See https://github.com/sourcegraph/sourcegraph/pull/10967#issuecomment-633835528

### CODE REVIEW

File: handbook/engineering/testing.md [hunk 1]
```diff
@@ -89,6 +89,12 @@ We use [Percy](https://percy.io/) to detect visual changes in Sourcegraph featur
 - [Distribution team](distribution/index.md) owns build and test infrastructure.
 - [Web team](web/index.md) owns any tests that are driven through the browser.
 
+## Conventions
+
+- **Naming tests in Go code.** We strive to for the following conventions when naming Go test functions:
```
Comment (rvantonder) [inline handbook/engineering/testing.md:94]:
```suggestion
- **Naming tests in Go code.** We strive to follow these conventions when naming Go test functions:
```

File: handbook/engineering/testing.md [hunk 2]
```diff
@@ -89,6 +89,12 @@ We use [Percy](https://percy.io/) to detect visual changes in Sourcegraph featur
 - [Distribution team](distribution/index.md) owns build and test infrastructure.
 - [Web team](web/index.md) owns any tests that are driven through the browser.
 
+## Conventions
+
+- **Naming tests in Go code.** We strive to follow these conventions when naming Go test functions:
+  - General rule: avoid using `_` immediately after `Test`. Instead, if testing a function like `fooBar`, the test is called TestFooBar and not `Test_fooBar`.
```
Comment (bobheadxi) [inline handbook/engineering/testing.md:95]:
```suggestion
  - General rule: avoid using `_` immediately after `Test`. Instead, if testing a function like `fooBar`, the test is called `TestFooBar` and not `Test_fooBar`. 
```

**Comment by nicksnyder** [comment]
Is this documented convention different than what is documented in the standard library? If not, can we link to the standard documentation instead of (or in addition to) trying to explain it ourselves?

https://golang.org/pkg/testing/#hdr-Examples

**Comment by rvantonder** [comment]
Oh that's interesting @nicksnyder, had no idea. The one comment I'd make is that the linked section refers to example functions rather than tests, but other than that it's seems to be exactly what we're going for. So something like "We use the same convention for naming test functions as example functions as described in the Go testing library [ink]"

**Pull request merged**